tests/trestle/core/commands/merge_test.py

```diff
@@ -0,0 +1,311 @@
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for trestle split command."""
+import argparse
+import os
+import shutil
+from pathlib import Path
+
+from dictdiffer import diff
+
+from tests import test_utils
+
+import trestle.oscal.catalog as oscatalog
+from trestle.core.commands.merge import MergeCmd
+from trestle.core.models.actions import CreatePathAction, RemovePathAction, WriteFileAction
+from trestle.core.models.elements import Element, ElementPath
+from trestle.core.models.file_content_type import FileContentType
+from trestle.core.models.plans import Plan
+from trestle.utils import fs
+from trestle.utils.load_distributed import load_distributed
+
+
+def test_merge_invalid_element_path(testdata_dir, tmp_trestle_dir):
+    """Test to make sure each element in -e contains 2 parts at least."""
+    cmd = MergeCmd()
+    args = argparse.Namespace(verbose=1, element='catalog')
+    assert cmd._run(args) == 1
+
+    args = argparse.Namespace(verbose=1, element='catalog.metadata')
+    test_utils.ensure_trestle_config_dir(tmp_trestle_dir)
+    test_data_source = testdata_dir / 'split_merge/step4_split_groups_array/catalogs'
+    catalogs_dir = Path('catalogs/')
+    mycatalog_dir = catalogs_dir / 'mycatalog'
+
+    # Copy files from test/data/split_merge/step4
+    shutil.rmtree(catalogs_dir)
+    shutil.copytree(test_data_source, catalogs_dir)
+
+    os.chdir(mycatalog_dir)
+    assert cmd._run(args) == 0
+
+
+def test_merge_plan_simple_case(testdata_dir, tmp_trestle_dir):
+    """Test '$mycatalog$ trestle merge -e catalog.back-matter'."""
+    # Assume we are running a command like below
+    # trestle merge -e catalog.back-matter
+    content_type = FileContentType.JSON
+    fext = FileContentType.to_file_extension(content_type)
+
+    # prepare trestle project dir with the file
+    test_utils.ensure_trestle_config_dir(tmp_trestle_dir)
+
+    test_data_source = testdata_dir / 'split_merge/step4_split_groups_array/catalogs'
+
+    catalogs_dir = Path('catalogs/')
+    mycatalog_dir = catalogs_dir / 'mycatalog'
+    catalog_dir = mycatalog_dir / 'catalog'
+
+    # Copy files from test/data/split_merge/step4
+    shutil.rmtree(catalogs_dir)
+    shutil.copytree(test_data_source, catalogs_dir)
+
+    os.chdir(mycatalog_dir)
+    catalog_file = Path(f'catalog{fext}')
+    catalog_dir = Path('catalog/')
+    back_matter_file = catalog_dir / f'back-matter{fext}'
```

Comment:
you would be missing the . from .json

Reply:
i'm actually surprised this works.

Reply:
`fext = FileContentType.to_file_extension(content_type)` includes the .